napari/layers/shapes/shapes.py

```diff
@@ -673,17 +673,23 @@ def data(self, data):
         self._data_view.slice_key = np.array(self._slice_indices)[
             self._slice_input.not_displayed
         ]
-        self.add(
+        self._add_shapes(
             data,
             shape_type=shape_type,
             edge_width=edge_widths,
             edge_color=edge_color,
             face_color=face_color,
             z_index=z_indices,
+            n_new_shapes=n_new_shapes,
         )
 
         self._update_dims()
-        self.events.data(value=self.data)
+        self.events.data(
+            value=self.data,
+            action=ActionType.CHANGE.value,
```

Comment:
why value, not just `ActionType.CHANGE`?

Reply:
In general, I prefer enums as it strictly point to possible values. Downstream packages could do the conversion on their side. But it may be done in follow-up PR.

Reply:
> it may be done in follow-up PR. 

I think it should, as this is already a valuable change and is consistent with our existing code.